bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieShell.java

```diff
@@ -1835,6 +1914,288 @@ public int runCmd(CommandLine cmdLine) throws Exception {
+        }
+
+        @Override
+        String getUsage() {
+            return CMD_TRIGGERAUDIT;
+        }
+
+        @Override
+        Options getOptions() {
+            return opts;
+        }
+
+        @Override
+        public int runCmd(CommandLine cmdLine) throws Exception {
+            ZooKeeper zk = null;
+            LedgerUnderreplicationManager underreplicationManager = null;
+            try {
+                zk = ZooKeeperClient.newBuilder()
+                        .connectString(bkConf.getZkServers())
+                        .sessionTimeoutMs(bkConf.getZkTimeout())
+                        .build();
+                LedgerManagerFactory ledgerManagerFactory = LedgerManagerFactory.newLedgerManagerFactory(bkConf, zk);
+                underreplicationManager = ledgerManagerFactory.newLedgerUnderreplicationManager();
+
+                if (!underreplicationManager.isLedgerReplicationEnabled()) {
+                    LOG.error("Autorecovery is disabled. So giving up!");
+                    return -1;
+                }
+                BookieSocketAddress auditorId = AuditorElector.getCurrentAuditor(bkConf, zk);
+                if (auditorId == null) {
+                    LOG.error("No auditor elected, though Autorecovery is enabled. So giving up.");
+                    return -1;
+                }
+
+                int previousLostBookieRecoveryDelayValue = underreplicationManager.getLostBookieRecoveryDelay();
+                LOG.info("Resetting LostBookieRecoveryDelay value: {}, to kickstart audit task",
+                        previousLostBookieRecoveryDelayValue);
+                underreplicationManager.setLostBookieRecoveryDelay(previousLostBookieRecoveryDelayValue);
+            } finally {
+                if (zk != null) {
+                    zk.close();
+                }
+            }
+            return 0;
+        }
+    }
+    
+    /**
+     * Command to trigger AuditTask by resetting lostBookieRecoveryDelay and then make sure the 
+     * ledgers stored in the bookie are properly replicated.
+     */
+    class DecommissionBookieCmd extends MyCommand {
+        Options lOpts = new Options();
+        int maxSleepTimeInBetweenChecks = 10 * 60 * 1000; // 10 minutes
+        int sleepTimePerLedger = 30 * 1000; // 30 secs
+        LedgerUnderreplicationManager underreplicationManager = null;
+        
+        DecommissionBookieCmd() {
+            super(CMD_DECOMMISSIONBOOKIE);
+        }
+
+        @Override
+        String getDescription() {
+            return "Force trigger the Audittask and make sure all the ledgers stored in the decommissioning bookie are replicated";
+        }
+
+        @Override
+        String getUsage() {
+            return CMD_DECOMMISSIONBOOKIE;
+        }
+
+        @Override
+        Options getOptions() {
+            return lOpts;
+        }
+
+        @Override
+        public int runCmd(CommandLine cmdLine) throws Exception {
+            ZooKeeper zk = null;
```

Comment:
you should put the decommision logic in bookkeeper admin. the shell here should be just a simple wrapper over that decommission method.

also we need tests for this behavior.

if you put the logic in Admin, you don't need to make DistributionSchedule public. These details should be hidden to the implementation.